android/app/src/main/java/org/fossasia/openevent/viewmodels/LocationsFragmentViewModel.java

```diff
@@ -5,28 +5,38 @@
 import android.arch.lifecycle.ViewModel;
 
 import org.fossasia.openevent.data.Microlocation;
-import org.fossasia.openevent.dbutils.LiveRealmData;
 import org.fossasia.openevent.dbutils.RealmDataRepository;
+import org.fossasia.openevent.dbutils.FilterableRealmLiveData;
 
 import java.util.List;
+import java.util.Locale;
 
-public class LocationsFragmentViewModel extends ViewModel {
+import io.reactivex.functions.Predicate;
+import io.realm.RealmResults;
 
-    private LiveData<List<Microlocation>> locations;
-    private RealmDataRepository realmRepo;
+public class LocationsFragmentViewModel extends ViewModel {
 
+    private RealmResults<Microlocation> realmResults;
+    private LiveData<List<Microlocation>> filteredLocations;
     private String searchText = "";
 
     public LocationsFragmentViewModel() {
-        realmRepo = RealmDataRepository.getDefaultInstance();
+        realmResults = RealmDataRepository.getDefaultInstance().getLocations();
     }
 
-    public LiveData<List<Microlocation>> getLocations() {
-        if (locations == null) {
-            LiveRealmData<Microlocation> microlocationLiveRealmData = RealmDataRepository.asLiveData(realmRepo.getLocations());
-            locations = Transformations.map(microlocationLiveRealmData, input -> input);
+    public LiveData<List<Microlocation>> getLocations(String searchText) {
+        if (!this.searchText.equals(searchText) || filteredLocations == null) {
+            setSearchText(searchText);
+            final String query = searchText.toLowerCase(Locale.getDefault());
+            Predicate<Microlocation> predicate = location -> location.getName()
+                    .toLowerCase(Locale.getDefault())
+                    .contains(query);
+            FilterableRealmLiveData<Microlocation> filterableRealmLiveData = RealmDataRepository.asFilterableLiveData(realmResults);
+            filterableRealmLiveData.setPredicate(predicate);
+            filterableRealmLiveData.filter();
+            filteredLocations = Transformations.map(filterableRealmLiveData, input -> input);
```

Comment:
This needs not be created again and again whenever there is a change in searchText, only the predicate will be changed and filter will be called, and that's why the only public method in the live data needs to be `filter(Predicate)` because these two methods will never be used without each other